colab_evaluation.py

```diff
@@ -0,0 +1,207 @@
+# coding=utf-8
+# Copyright 2020 The Google Research Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Version of evaluation utilities intended for use in lower memory environments, such as colab."""
+
+import numpy as np
+import pandas as pd
+import sklearn
+import tqdm
+
+import inference
+import utils
+import evaluation
+
+
+def batch_inferences(iterator, batch_size):
+    """Yield batches of seq_ids and predictions matrices from an iterator."""
+    counter = 0
+    predictions = []
+    seq_ids = []
+    while True:
+        try:
+            inference = next(iterator)
+        except StopIteration:
+            yield seq_ids, np.vstack(predictions)
+            return
+        seq_ids.append(inference[0])
+        predictions.append(inference[1])
+        counter += 1
+        if counter == batch_size:
+            yield seq_ids, np.vstack(predictions)
+            predictions = []
+            seq_ids = []
+            counter = 0
+
+
+def batched_inferences_from_files(shard_names, batch_size=100):
+    """Iterate through TFRecord files of inferences and yield batches."""
+    for file_name in tqdm.tqdm(shard_names, position=0):
+        inference_iterator = inference.parse_shard(file_name)
+        batched_iterator = batch_inferences(inference_iterator, batch_size)
+        while True:
+            try:
+                yield next(batched_iterator)
+            except StopIteration:
+                break
+
+
+def batched_inferences_from_dir(shard_dir_path, batch_size=100):
+    """Iterate through directory of inference TFRecord files and yield batches."""
+    files_to_process = utils.absolute_paths_of_files_in_dir(shard_dir_path)
+    return batched_inferences_from_files(files_to_process, batch_size)
+
+
+def make_tidy_df_from_seq_names_and_prediction_array(
+        sequence_names,
+        predictions_array,
+        vocab,
+        min_decision_threshold=1e-20):
+    """Given a list of sequences and a matrix of prediction values, yield a tidy dataframe of predictions."""
+    up_ids = []
+    labels = []
+    values = []
+
+    for i in range(len(sequence_names)):
+        up_id = sequence_names[i]

```

Comment:
Ah neat. In one sense I find the current version clearer because it's explicit about the dimensionality (but I also like the elegance of what you have).